sdk/nodejs/cmd/run-plugin/run.ts

```diff
@@ -0,0 +1,304 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The tsnode import is used for type-checking only. Do not reference it in the emitted code.
+import * as tsnode from "ts-node";
+import * as fs from "fs";
+import * as util from "util";
+import * as minimist from "minimist";
+import * as path from "path";
+import * as tsutils from "../../tsutils";
+import { ResourceError, RunError } from "../../errors";
+import * as log from "../../log";
+import * as settings from "../../runtime/settings";
+import * as stack from "../../runtime/stack";
+
+// Keep track if we already logged the information about an unhandled error to the user..  If
+// so, we end with a different exit code.  The language host recognizes this and will not print
+// any further messages to the user since we already took care of it.
+//
+// 32 was picked so as to be very unlikely to collide with any of the error codes documented by
+// nodejs here:
+// https://github.com/nodejs/node-v0.x-archive/blob/master/doc/api/process.markdown#exit-codes
```

Comment:
Again I think https://nodejs.org/api/process.html#process_exit_codes would be better here.